src/test/java/uk/gov/pay/ledger/transaction/dao/TransactionDaoIT.java

```diff
@@ -43,76 +43,77 @@ public void shouldInsertTransaction() {
-        assertThat(retrievedTransaction.getEmail(), is(transactionEntity.getEmail()));
-        assertThat(retrievedTransaction.getCardholderName(), is(transactionEntity.getCardholderName()));
-        assertThat(retrievedTransaction.getTransactionDetails(), containsString("\"external_metadata\": {\"key1\": \"value1\", \"anotherKey\": {\"nestedKey\": \"value\"}}"));
-        assertThat(retrievedTransaction.getCreatedDate(), is(transactionEntity.getCreatedDate()));
-        assertThat(retrievedTransaction.getTransactionDetails().contains(fixture.getLanguage()), is(true));
-        assertThat(retrievedTransaction.getTransactionDetails().contains(fixture.getReturnUrl()), is(true));
-        assertThat(retrievedTransaction.getTransactionDetails().contains(fixture.getPaymentProvider()), is(true));
-        assertThat(retrievedTransaction.getTransactionDetails().contains(fixture.getCardDetails().getBillingAddress().getAddressLine1()), is(true));
-        assertThat(retrievedTransaction.getEventCount(), is(transactionEntity.getEventCount()));
-        assertThat(retrievedTransaction.getCardBrand(), is(transactionEntity.getCardBrand()));
-        assertThat(retrievedTransaction.getLastDigitsCardNumber(), is(transactionEntity.getLastDigitsCardNumber()));
-        assertThat(retrievedTransaction.getFirstDigitsCardNumber(), is(transactionEntity.getFirstDigitsCardNumber()));
+        assertTransactionEntity(retrievedTransaction, fixture);
         assertThat(retrievedTransaction.getNetAmount(), is(transactionEntity.getNetAmount()));
         assertThat(retrievedTransaction.getTotalAmount(), is(transactionEntity.getTotalAmount()));
         assertThat(retrievedTransaction.getFee(), is(transactionEntity.getFee()));
-        assertThat(retrievedTransaction.getTransactionType(), is(transactionEntity.getTransactionType()));
-        assertThat(retrievedTransaction.getRefundAmountAvailable(), is(transactionEntity.getRefundAmountAvailable()));
-        assertThat(retrievedTransaction.getRefundAmountRefunded(), is(transactionEntity.getRefundAmountRefunded()));
-        assertThat(retrievedTransaction.getRefundStatus(), is(transactionEntity.getRefundStatus()));
-        assertThat(retrievedTransaction.getGatewayTransactionId(), is(transactionEntity.getGatewayTransactionId()));
         assertThat(retrievedTransaction.isLive(), is(true));
     }
 
     @Test
-    public void shouldRetrieveTransactionByExternalIdAndGatewayAccountId() {
+    public void shouldRetrieveTransactionByExternalIdAndGatewayAccount() {
         TransactionFixture fixture = aTransactionFixture()
                 .withDefaultCardDetails()
-                .withTransactionType("PAYMENT")
                 .withExternalMetadata(ImmutableMap.of("key1", "value1", "anotherKey", ImmutableMap.of("nestedKey", "value")))
+                .withTransactionType("PAYMENT")
+                .withGatewayTransactionId("gateway_transaction_id")
```

Comment:
are specifying transactionType and gatewayTransactionId relevant to this PR?

Reply:
No,  they're not (copy+paste+refactor didn't get to remove those two lines).
I can remove them.